### ISSUE TITLE
Restart / Reload Session

### ISSUE THREAD

**Issue by wdpronovost** [warpdotdev/Warp#2439]
### Describe the solution you'd like?

```text
When using Anaconda, I have to open a new Tab in order to verify if the 'activate' or 'deactivate' command was successful. It would be helpful to have a way to reload / restart the current session. A key command (like CMD + R) would be great as it is how we reload a browser session.
```

### Is your feature request related to a problem? Please describe.

```text
Not so much a problem, just an inconvenience in ensuring I am in the correct environment.
```

### Additional context

iTerm2 called it Restart Session so that where I am getting the terms I am using.

### How important is this feature to you?

2

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by dannyneira** [comment]
Cool idea, thanks for submitting it. afaik the only way to refresh is open a new tab, it'll default to the same cxd as the focused tab.

Please :+1:  and subscribe to this request as it helps us gauge interest. We will post any updates here.

**Comment by sAnti09** [comment]
This is also important to me as sometimes I would not successfully connect via ssh and it freezes and nothing could be done with the tab. So what I do is close the tab and split the pane again.

It would be nice to just have an option to restart the tab.

**Comment by jpmadrigal07** [comment]
I agree with this one. By simply right-clicking in the terminal, we need to be able to see a `Refresh Session` option. This is helpful for froze screen.
 
![image](https://github.com/warpdotdev/Warp/assets/20082606/19a0f9ba-3450-452f-bede-c6041cc302ab)

**Comment by iamgbayer** [comment]
+1 for this

**Comment by lafkpages** [comment]
Would this also work for when you edit your `.zshrc`, `.bashrc`, etc and want to apply your changes?

**Comment by SwiftWinds** [comment]
yeah that's the thing. If I simply do `exec zsh` to reload zsh session, Warp prompt goes away and I'm left with my oh my zsh prompt:

<img width="687" alt="image" src="https://github.com/user-attachments/assets/d22642ae-10ac-4326-a1f3-4ba2d45f970a">

If a keyboard shortcut to reload the sesh (say, cmd + r) were added, I would be overjoyed! :)

**Comment by KRRT7** [comment]
+1 on this

**Comment by GrantMStevens** [comment]
This is one of the things I miss most from my iTerm2 setup. I like Warp, but I run into issues that this would solve often enough that it gets pretty annoying.

**Comment by sAnti09** [comment]
Any updates on this? :)

**Comment by ginevz** [comment]
+1 this is a must have feature

**Comment by leoboyerbx** [comment]
My current workaround is by having a alias rl="exec zsh" in my .zshrc. Then in the settings > warpify, I add the command "rl" to the list.

When I want to reload, it type rl, and have to press ^+I, that reloads a shell and adds warp prompt.

But yes +1 on having this built-in !

**Comment by ns-rpaikaray** [comment]
It should support bydefault as like iterm2 terminal.Any update on this ?

**Comment by rishitank** [comment]
This should be a common sense feature, any update on this?

**Comment by Legoless** [comment]
Hi @oss-maintainers — I've opened PR #14001 implementing a Reload Shell action for this request. Could you re-evaluate this issue's readiness and mark it `ready-to-implement` (or `ready-to-spec` if you'd prefer a spec PR first) so Oz can proceed with review? Happy to adjust the approach based on your feedback. Thanks!
